forte/data/data_store.py

```diff
@@ -638,7 +690,78 @@ def add_annotation_raw(
         # A reference to the entry should be store in both self.__elements and
         # self.__tid_ref_dict.
         entry = self._new_annotation(type_name, begin, end, tid)
-        if not allow_duplicate and type_name in self.__elements:
+        if not allow_duplicate:
+            tid_search_result = self._get_existing_entry_tid(entry)
+            # if found existing entry
+            if tid_search_result != -1:
+                return tid_search_result
+
+        return self._add_entry_raw(Annotation, type_name, entry)
+
+    def add_audio_annotation_raw(
+        self,
+        type_name: str,
+        begin: int,
+        end: int,
+        tid: Optional[int] = None,
+        allow_duplicate=True,
+    ) -> int:
+
+        r"""This function adds an audio annotation entry with ``begin`` and
+        ``end``
```

Comment:
could consider reorganizing here to make the text has natural line breaks. Note that people would also read the source code directly so documentation is helpful at that time.